### DISCUSSION TITLE
GraphQL: Allow Config To Whitelist only Certain Domains for CORS

### DISCUSSION THREAD

**Discussion by edmotech** [craftcms/cms#8728] [Ideas] [answered]
### Description

It would be great if you could specify a whitelist for the GraphQL CORS to lock down the api to certain domains.

### Additional info

- Craft version: 3.3.0.1
- PHP version: 7.2
- Database driver & version: mysql 5.6
- Plugins & versions:

**Comment by AugustMiller** [comment]
This seems like something that could be tied to the Schema/Token system, sort of like Google Cloud API keys? 🤔

**Comment by simeon-smith** [comment]
@AugustMiller that would definitely be an interesting idea and a cool feature. My primary concern (as posted under the edmotech user) is simply to allow locking down the API via a CORS whitelist.

**Comment by timkelty** [comment, marked as answer]
This is implemented with the general config setting: [`allowedGraphqlOrigins`](https://craftcms.com/docs/4.x/config/general.html#allowedgraphqlorigins)
